### PR TITLE
Fixing image splitting

### PR DESCRIPTION
**Pull request by iron768**
We fixed image splitting and an error for CLI client

### CODE REVIEW

**Review by lettucegoblin — COMMENTED** [review]
fucking incredible

File: src/main/java/org/fibsters/BufferedImageTypeAdapter.java [hunk 1]
```diff
@@ -54,8 +55,11 @@ public void write(JsonWriter jsonWriter, BufferedImage image) throws IOException
 
     @Override
     public BufferedImage read(JsonReader jsonReader) throws IOException {
+        if (jsonReader.peek() == JsonToken.NULL) {
+            jsonReader.nextNull();
```
Comment (lettucegoblin) [inline src/main/java/org/fibsters/BufferedImageTypeAdapter.java:59]: 5 hours of confusion solved in one line.

File: src/main/java/org/fibsters/ComputeJobPoolImpl.java [hunk 2]
```diff
@@ -130,32 +130,34 @@ private void processJob(ComputeJob job) {
         int[] subJobs = job.getInputPayload().getPayloadDataParsed(); // ex [1, 10, 25]
 
         int[] threadsPerSubjob = this.handleThreadPooling(subJobs, numThreads);
-
         int threadCount = 0;
 
-        for (int i = 0; i < threadsPerSubjob.length; i++) {
-            int threadGroupSize = subJobs[i] / threadsPerSubjob[i];
+            for (int i = 0; i < threadsPerSubjob.length; i++) {
```
Comment (lettucegoblin) [inline src/main/java/org/fibsters/ComputeJobPoolImpl.java]: Fix the spacing pls

File: src/main/java/org/fibsters/OutputPayloadImpl.java [hunk 3]
```diff
@@ -32,7 +34,7 @@ public OutputPayloadImpl(int index, InputPayloadImpl inputPayload, ComputeJobSta
 
         // need to parse the height/width
         // temporarily set it as whatever
-        this.fibSpiralResult = new BufferedImage(2000, 2000, BufferedImage.TYPE_INT_RGB);
+        this.fibSpiralResults = new CopyOnWriteArrayList<>();
```
Comment (lettucegoblin) [inline src/main/java/org/fibsters/OutputPayloadImpl.java:37]: thread safety is best safety

**Review by lettucegoblin — APPROVED** [review]
LGTM

File: src/main/java/org/fibsters/BufferedImageTypeAdapter.java (hunk 1 above)
Comment (ndg8743) [reply src/main/java/org/fibsters/BufferedImageTypeAdapter.java:59]: image

**Pull request merged**